### PR TITLE
Update persistent-storage-csi-snapshots-restore.adoc

### PR DESCRIPTION
**Pull request by JuozasA**
removing API version from apiGroup #29919 🪣

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@JuozasA Thanks for submitting this! 

However I have a question on what OCP version(s) this applies to. Is it really only 4.7? Or does it also apply going forward to 4.8, and/or potentially back further than 4.7?

If this change needs to be made in the newest version too (4.8+), can you please make this update against the master branch, instead of enterprise-4.7? Then once it's merged, we'll cherry pick it back to all other appropriate versions. Thanks!

**Comment by bergerhoffer** [comment]
@bobfuru or @lpettyjo Can you recommend a storage QE to review this update? Thanks!

**Comment by bobfuru** [comment]
@duanwei33 PTAL for QE with these questions for you:

If this fix is correct, it should apply to 4.7+, correct? (4.5 & 4.6 docs do not have the `v1` version in `dataSource.apiGroup`).

Also, there are two other occurrences where this fix should also be applied in the **Storage -> CSI volume snapshots** doc:
- `modules/persistent-storage-csi-snapshots-operator.adoc` (line 14)

    >  During {product-title} installation, the CSI Snapshot Controller Operator creates the following snapshot custom resource definitions (CRDs) in the `snapshot.storage.k8s.io/v1` API group:

- `modules/persistent-storage-csi-snapshots-restore.adoc` (line 42) 

    > <3> Must be set to the `snapshot.storage.k8s.io/v1` value.

Do you agree?

(Thanks @JuozasA for submitting this. Once QE reviews, I agree with @bergerhoffer that these changes should be made against master branch and then we can CP to 4.7 and 4.8.)

**Comment by duanwei33** [comment]
@bobfuru @JuozasA 

> If this fix is correct, it should apply to 4.7+, correct? (4.5 & 4.6 docs do not have the `v1` version in `dataSource.apiGroup`).
[duanwei33] Yes, this should apply to 4.7+.
> 
> Also, there are two other occurrences where this fix should also be applied in the **Storage -> CSI volume snapshots** doc:
> 
> * `modules/persistent-storage-csi-snapshots-operator.adoc` (line 14)
>   > During {product-title} installation, the CSI Snapshot Controller Operator creates the following snapshot custom resource definitions (CRDs) in the `snapshot.storage.k8s.io/v1` API group:
[wduan] Sorry I did not find this module in OCP doc, could you give me a doc link? To be honest, I see in our doc API section, it also take the version after the api group in https://docs.openshift.com/container-platform/4.7/rest_api/index.html, so I'm not sure if need to modify here. (Not like in yaml file)
@qinpingli WDYT?   

> * `modules/persistent-storage-csi-snapshots-restore.adoc` (line 42)
>   > <3> Must be set to the `snapshot.storage.k8s.io/v1` value.
[duanwei33] Yes, in this module, this should be updated as well.
> 
> Do you agree?
> 
> (Thanks @JuozasA for submitting this. Once QE reviews, I agree with @bergerhoffer that these changes should be made against master branch and then we can CP to 4.7 and 4.8.)

**Comment by bobfuru** [comment]
Thanks, @duanwei33 

> > * `modules/persistent-storage-csi-snapshots-operator.adoc` (line 14)
> >   > During {product-title} installation, the CSI Snapshot Controller Operator creates the following snapshot custom resource definitions (CRDs) in the `snapshot.storage.k8s.io/v1` API group:
> 
> [duanwei33] Sorry I did not find this module in OCP doc, could you give me a doc link? To be honest, I see in our doc API section, it also take the version after the api group in https://docs.openshift.com/container-platform/4.7/rest_api/index.html, so I'm not sure if need to modify here. (Not like in yaml file)

The doc link you ask about is here: https://docs.openshift.com/container-platform/4.7/storage/container_storage_interface/persistent-storage-csi-snapshots.html#volume-snapshot-crds

But I agree, based on the API docs you referenced, it looks like we should leave this occurrence as it is currently.

**Comment by duanwei33** [comment]
Except for the following part, others look good to me.

> modules/persistent-storage-csi-snapshots-restore.adoc (line 42)
> <3> Must be set to the snapshot.storage.k8s.io/v1 value.
> need to update to 
> <3> Must be set to the snapshot.storage.k8s.io value.

**Comment by duanwei33** [comment]
Except for the following part, others look good to me.

> modules/persistent-storage-csi-snapshots-restore.adoc (line 42)
> <3> Must be set to the snapshot.storage.k8s.io/v1 value.
> need to update to 
> <3> Must be set to the snapshot.storage.k8s.io value.

File: modules/persistent-storage-csi-snapshots-restore.adoc [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
   dataSource:
     name: mysnap <1>
     kind: VolumeSnapshot <2>
-    apiGroup: snapshot.storage.k8s.io/v1 <3>
+    apiGroup: snapshot.storage.k8s.io <3>
```
Comment (bobfuru) [inline modules/persistent-storage-csi-snapshots-restore.adoc:33]: Please also remove `/v1` from the annotated note on line 42. Other than that, this LGTM.

**Comment by qinpingli** [comment]
> > Also, there are two other occurrences where this fix should also be applied in the **Storage -> CSI volume snapshots** doc:
> > 
> > * `modules/persistent-storage-csi-snapshots-operator.adoc` (line 14)
> >   > During {product-title} installation, the CSI Snapshot Controller Operator creates the following snapshot custom resource definitions (CRDs) in the `snapshot.storage.k8s.io/v1` API group:
> 
> [duanwei33] Sorry I did not find this module in OCP doc, could you give me a doc link? To be honest, I see in our doc API section, it also take the version after the api group in https://docs.openshift.com/container-platform/4.7/rest_api/index.html, so I'm not sure if need to modify here. (Not like in yaml file)
> @qinpingli WDYT?
Sorry, I don't understand the issue totally. Usually, we call `snapshot.storage.k8s.io/v1` as apiVersion, call `snapshot.storage.k8s.io` as apiGroup.

**Comment by bobfuru** [comment]
Based on @qinpingli's [comment](https://github.com/openshift/openshift-docs/pull/29924#issuecomment-791080598), we need a more definitive yes or no to whether this is correct as it currently describes in the doc:
> `modules/persistent-storage-csi-snapshots-operator.adoc` (line 14)
> During {product-title} installation, the CSI Snapshot Controller Operator creates the following snapshot custom resource definitions (CRDs) in the snapshot.storage.k8s.io/v1 API group

**Comment by bobfuru** [comment]
This fix is applied in https://github.com/openshift/openshift-docs/pull/30795
Closing issue. Thanks for your help, @JuozasA!

**Pull request closed without merging**